### PR TITLE
nghttp2: 1.51.0 -> 1.54.0

### PR DESCRIPTION
**Pull request by Izorkin**
###### Description of changes
Update nghttp2 to version 1.54.0.
Cleanup and add options to allow http3 support.
In 1.52.0 removed deprecated python bindings.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/nghttp2/default.nix [hunk 1]
```diff
@@ -27,41 +27,39 @@
...
   };
 
-  outputs = [ "bin" "out" "dev" "lib" ]
-    ++ lib.optionals (enablePython) [ "python" ];
+  outputs = [ "out" "dev" "lib" "doc" "man" ];
 
   nativeBuildInputs = [ pkg-config ]
-    ++ lib.optionals (enableApp) [ installShellFiles ]
-    ++ lib.optionals (enablePython) [ python3Packages.cython ];
+    ++ lib.optionals (enableApp) [ installShellFiles ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nghttp2/default.nix:45]:
```suggestion
    ++ lib.optionals enableApp [ installShellFiles ];
```
all of those are noops

```suggestion
    ++ lib.optional enableApp installShellFiles;
```
or so

File: pkgs/development/libraries/nghttp2/default.nix [hunk 2]
```diff
@@ -27,41 +27,39 @@
...
   configureFlags = [
     "--disable-examples"
     (lib.enableFeature enableApp "app")
-  ] ++ lib.optionals (enableAsioLib) [ "--enable-asio-lib" "--with-boost-libdir=${boost}/lib" ]
-    ++ lib.optionals (enablePython) [
-      "--enable-python-bindings"
-      "--with-cython=${python3Packages.cython}/bin/cython"
-    ];
+  ] ++ lib.optionals (enableHttp3) [
+    "--enable-http3"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nghttp2/default.nix:61]:
```suggestion
    (lib.enableFeature enableHttp3 "http3")
```

**Review by SuperSandro2000 — COMMENTED** [review]
Maybe squash the two nghttp2 commits. Also please add an entry to python-aliases.nix

**Comment by Izorkin** [comment]
@SuperSandro2000 thaks, fixed PR.

**Comment by jktjkt** [comment]
Upstream also removed the C++ `boost-asyncio` wrapper which got moved to [an extra repo](https://github.com/nghttp2/nghttp2-asio). Can you please add that as a package? Here's what works for me in my `foo.nix`:
```nix
  nghttp2_asio = pkgs.stdenv.mkDerivation rec {
    pname = "nghttp2-asio";
    version = "0.0.20230712.0";

    src = pkgs.fetchFromGitHub {
      owner = "nghttp2";
      repo = "nghttp2-asio";
      rev = "e877868abe06a83ed0a6ac6e245c07f6f20866b5";
      sha256 = "sha256-XQXRHLz0kvaIQq1nbqkJnETHR51FXMB1P9F/hQeZh6A=";
    };

    nativeBuildInputs = [
      cmake
      nghttp2
      openssl
      boost
    ];
  };
```

**Pull request merged**